integrations/nvidia/tests/test_text_embedder.py

```diff
@@ -105,6 +126,28 @@ def from_dict(self, monkeypatch):
         assert component.suffix == "suffix"
         assert component.truncate == "START"
 
+    @pytest.mark.usefixtures("mock_local_models")
+    def test_run_default_model(self):
+        embedder = NvidiaTextEmbedder(api_url="http://localhost:8080/v1", api_key=Secret.from_token("fake-api-key"))
+
+        assert embedder.model is None
+
+        with pytest.warns(UserWarning):
```

Comment:
to ensure the expected warning is raised, please name the warning and check its message. for instance -

```
    with pytest.warns(UserWarning) as record:
        ...
    assert len(record) == 1
    assert "Default model set to" in str(record[0].message)
```